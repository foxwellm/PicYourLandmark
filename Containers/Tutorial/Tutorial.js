import React, { Component } from 'react';
import { StyleSheet, Text, View, ImageBackground } from 'react-native';
import { Icon } from 'react-native-elements';

export class Tutorial extends Component {
  render() {
    return (
      <View style={styles.viewContainer}>
        <ImageBackground source={require('../../assets/pyramid.jpg')} style={styles.imageBackground}>
          <View style={styles.overlay} />
          <Text style={styles.bannerText}>TUTORIAL</Text>
          <Icon style={styles.bannerIcon} color="white" name="book" type="font-awesome" size={40} />
        </ImageBackground>
        <View style={styles.tutorialInfo}>
          <Text>Tutorial text</Text>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  viewContainer: {
    flex: 3,
    width: '100%',
    display: 'flex',
    flexDirection: 'column',
  },
  imageBackground: {
    flex: 1.5,
    width: '100%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  overlay: {
    position: 'absolute',
    top: 0,
    width: '100%',
    height: '100%',
    backgroundColor: 'black',
    opacity: 0.4,
  },
  tutorialInfo: {
    flex: 3
  },
  bannerText: {
    color: 'white',
    paddingBottom: '3%',
    fontSize: 18,
    fontWeight: '600',
  }
});

export default Tutorial;