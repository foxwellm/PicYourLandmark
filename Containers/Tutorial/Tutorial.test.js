import React from 'react';
import { Tutorial } from './Tutorial';
import { shallow } from 'enzyme';
import { TouchableOpacity } from 'react-native';

describe('Tutorial', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Tutorial />)
  });

  it('should match the snapshot with the correct data passed in when on User profile page', () => {
    expect(wrapper).toMatchSnapshot()
  });
});
