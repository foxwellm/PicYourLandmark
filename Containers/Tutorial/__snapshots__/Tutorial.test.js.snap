// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tutorial should match the snapshot with the correct data passed in when on User profile page 1`] = `
<Component
  style={
    Object {
      "display": "flex",
      "flex": 3,
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <ImageBackground
    source={1}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flex": 1.5,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Component
      style={
        Object {
          "backgroundColor": "black",
          "height": "100%",
          "opacity": 0.4,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    />
    <Component
      style={
        Object {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "600",
          "paddingBottom": "3%",
        }
      }
    >
      TUTORIAL
    </Component>
    <Themed.Icon
      color="white"
      name="university"
      size={40}
      type="font-awesome"
    />
  </ImageBackground>
</Component>
`;
